server/src/main/java/kr/codesquad/library/service/BookSearchService.java

```diff
@@ -0,0 +1,73 @@
+package kr.codesquad.library.service;
+
+import kr.codesquad.library.domain.book.Book;
+import kr.codesquad.library.domain.book.BookRepository;
+import kr.codesquad.library.domain.book.response.BookResponse;
+import kr.codesquad.library.domain.book.response.BooksByCategoryResponse;
+import kr.codesquad.library.domain.category.CategoryRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static kr.codesquad.library.domain.book.BookVO.PAGESIZE;
+
+@RequiredArgsConstructor
+@Service
+public class BookSearchService {
+
+    private final CategoryRepository categoryRepository;
+    private final BookRepository bookRepository;
+
+    @Transactional(readOnly = true)
+    public List<BooksByCategoryResponse> findMainBooks() {
+        List<BooksByCategoryResponse> mainBooks = new ArrayList<>();
+        for (int i = 0; i < categoryRepository.count(); i++) {
+            mainBooks.add(findTop6BooksAndCategoryById((long) i + 1));
+        }
+        return mainBooks;
+    }
+
+    @Transactional(readOnly = true)
+    public BooksByCategoryResponse findTop6BooksAndCategoryById(Long categoryId) {
+        return BooksByCategoryResponse.builder()
+                .categoryId(categoryId)
+                .categoryTitle(categoryRepository.findById(categoryId).orElseThrow(()
+                        -> new IllegalStateException("해당하는 " + categoryId + "의 카테고리는 없습니다.")).getTitle())
+                .bookCount(bookRepository.countByCategoryId(categoryId))
+                .books(findTop6BooksByCategory(categoryId))
+                .build();
+    }
```

Comment:
가독성을 위해 findById() 는 분리하는게 좋다고 생각해요.

```suggestion
    @Transactional(readOnly = true)
    public BooksByCategoryResponse findTop6BooksAndCategoryById(Long categoryId) {
        String title = categoryRepository.findById(categoryId).orElseThrow(()
                        -> new IllegalStateException("해당하는 " + categoryId + "의 카테고리는 없습니다.")).getTitle()
        return BooksByCategoryResponse.builder()
                .categoryId(categoryId)
                .categoryTitle(title)
                .bookCount(bookRepository.countByCategoryId(categoryId))
                .books(findTop6BooksByCategory(categoryId))
                .build();
    }
```

Reply:
오히려 이게 더 가독성이 나오겠군요 감사합니다 고쳐보도록 할게요. 근데 아마 이것도 중복코드가 많아서 ExceptionHandler를 사용하여 리팩토링할 예정이라 많이 달라질듯 합니다.